kubernetes/resource_kubernetes_service_account.go

```diff
@@ -237,6 +261,9 @@ func diffObjectReferences(origOrs []api.ObjectReference, ors []api.ObjectReferen
 }
 
 func resourceKubernetesServiceAccountRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	var diagMessages []diag.Diagnostic
```

Comment:
you could put this in the function signature instead, like this:

```
func resourceKubernetesServiceAccountRead(...) (diag diag.Diagnostics)
```

Reply:
I have updated this function and moved the server version verification to the end of it. Should be more clean and readable now.